java/core/src/main/java/sleeper/core/statestore/CommitRequestType.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.statestore;
+
+/**
+ * Represents the different types of state store commit requests that can be performed.
+ */
+public enum CommitRequestType {
+    COMPACTION,
+    INGEST
```

Comment:
It's conceivable that we might want to add other types of update to do with compaction here, eg. file assignment. Can we be a bit more descriptive about what the state store update is doing? This could be something like COMPACTION_FINISHED, as opposed to COMPACTION_ASSIGN_FILES? The ingest one could be INGEST_ADD_FILES or just ADD_FILES.